### ISSUE TITLE
Cannot find module 'insert-css'

### ISSUE THREAD

**Issue by mattdesl** [sheetify/sheetify#51]
I have the following:

```js
const sf = require('sheetify')
const prefix = sf`
  h1 {
    text-align: center;
  }
`
console.log(prefix)
```

I'm guessing it gets transformed into some code that includes `require('insert-css')` – this won't work unless my project explicitly installs that as a dependency. Due to npm3 flattening, this might not be immediately obvious in some projects. :smile:

**Comment by yoshuawuyts** [comment]
yes, this is correct - unless you're exorcising the CSS to an external file (or stream) it will inject an `insert-css` statement to append the CSS to the head of the document.

npm3 dedupe is kinda tricky; do you reckon setting `insert-css` as an (optional) peerdependency might resolve it? I'll add this to the FAQ regardless, as I suspect others might run into this too

**Comment by mattdesl** [comment]
Alternatively, you can bundle the `insert-css` module on pre-publish like I did [here](https://github.com/Jam3/tap-dev-tool/blob/master/index.js#L6)... But I think peer dependency might be better for a couple reasons:

- insert-css is pretty much a frozen module, so unlikely to run into the usual woes of dealing with peer deps
- a quick glance at the sheetify org tells me you might want to keep the insert-css require statement there so it can be found in other transforms/plugins that might optimize the CSS out of the bundle

**Comment by ahdinosaur** [comment]
yeah, it used to be a dependency that was required by absolute path, i'd prefer we make it a peer dependency so something like [a separate exorcising transform](https://github.com/sheetify/bundle-cssify) that uses a module name possible.

**Comment by yoshuawuyts** [comment]
hmmm, I believe Hugh wanted to make it a fixed dep inside `sheetify` again. I'm a bit torn between the two approaches haha

@ahdinosaur what do you mean by "uses a module name?" I def see why exorcising `'insert-css'` calls by another plugin is interesting, regardless that CSS can already be captured by a stream already.

**Comment by ahdinosaur** [comment]
@yoshuawuyts we insert the css using a specific module named `insert-css` (not an absolute path as before), so we can easily write a transform to capture all css being inserted with [`static-module`](https://github.com/substack/static-module). the problem with capturing it via a stream is that it requires more than just some additions to your package.json `browserify.transform` config.

**Comment by yoshuawuyts** [comment]
ah yeah, that's fully fair. I think the concensus is to make it a peer dependency then - PRs welcome, though I'm also happy to write in a couple of hours (watching deadpool first ✌️)

**Comment by yoshuawuyts** [comment]
oh god, this is super annoying - didn't come around to resolving it yest, should be fixed tho

**Comment by yoshuawuyts** [comment]
Fixed in `v4.0.7`. Thanks!
